### PR TITLE
Address coverity hits

### PR DESCRIPTION
**Pull request by nupuruttarwar**
Remove unused variable and return error if meter and register entry doesn't have valid index

### CODE REVIEW

File: stratum/hal/lib/tdi/tdi_table_manager.cc [hunk 1]
```diff
@@ -995,7 +995,11 @@ ::util::Status TdiTableManager::WriteRegisterEntry(
   absl::optional<uint32> register_index;
   if (register_entry.has_index()) {
     register_index = register_entry.index().index();
+  } else {
+    RETURN_ERROR(ERR_INVALID_PARAM) << "Register entry does not have "
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_table_manager.cc]: Would it be clearer to say "Invalid register entry index"?

"does not have a valid index" is a roundabout way of saying it

File: stratum/hal/lib/tdi/tdi_table_manager.cc [hunk 2]
```diff
@@ -1100,7 +1104,10 @@ ::util::Status TdiTableManager::WriteMeterEntry(
   absl::optional<uint32> meter_index;
   if (meter_entry.has_index()) {
     meter_index = meter_entry.index().index();
+  } else {
+    RETURN_ERROR(ERR_INVALID_PARAM) << "Meter entry does not have valid index";
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_table_manager.cc]: "Invalid meter entry index"?

**Review by ffoulkes — APPROVED** [review]
LGTM with suggestions

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**